### PR TITLE
Attention model evaluation PIDML adjustments

### PR DESCRIPTION
**Pull request by mytkom**

### CODE REVIEW

File: Tools/PIDML/pidMLEffAndPurProducer.cxx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    return nSigma;
+  }
+
+  bool IsNSigmaAccept(const BigTracks::iterator& track, nSigma_t& nSigma)
+  {
+    int sign = cfgPid > 0 ? 1 : -1;
+    if (track.sign() != sign)
+      return false;
+
+    if (track.pt() <= cfgTofPtCut || (track.tofSignal() - kTOFMissingSignal) < kEpsilon) {
```
Comment (WildJellybear) [inline Tools/PIDML/pidMLEffAndPurProducer.cxx]: dito

File: Tools/PIDML/pidMLBatchEffAndPurProducer.cxx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        nSigma.tof = track.tofNSigmaKa();
+        nSigma.tpc = track.tpcNSigmaKa();
+        break;
+      case 2212: // proton
+        nSigma.tof = track.tofNSigmaPr();
+        nSigma.tpc = track.tpcNSigmaPr();
+        break;
+    }
+
+    if (track.pt() < nSigmaTofPtCut || (track.tofSignal() - tofMissing) < eps) {
```
Comment (WildJellybear) [inline Tools/PIDML/pidMLBatchEffAndPurProducer.cxx]: add abs to tofSignal check

File: Tools/PIDML/pidMLEffAndPurProducer.cxx (hunk 1 above)
Comment (WildJellybear) [inline Tools/PIDML/pidMLEffAndPurProducer.cxx]: (not important) keep one convention - in batch there is exclusive pt cut, just to be sure both scripts would give same results

File: Tools/PIDML/pidOnnxModel.h [hunk 3]
```diff
@@ -179,36 +169,48 @@ struct PidONNXModel {
...
+      float scaledTRDSignal = (track.trdSignal() - mScalingParams.at("fTRDSignal").first) / mScalingParams.at("fTRDSignal").second;
+      inputValues.push_back(scaledTRDSignal);
+      inputValues.push_back(track.trdPattern());
+    } else {
+      inputValues.push_back(std::numeric_limits<float>::quiet_NaN());
+      inputValues.push_back(std::numeric_limits<float>::quiet_NaN());
+    }
 
-    if (mDetector >= kTPCTOF) {
+    if ((track.tofSignal() - kTOFMissingSignal) > kEpsilon) {
```
Comment (WildJellybear) [inline Tools/PIDML/pidOnnxModel.h]: dito

File: Tools/PIDML/pidMLBatchEffAndPurProducer.cxx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file pidMLBatchEffAndPurProducer
+/// \brief Batch PID execution task. It produces derived data needed for ROOT script which
+/// generates efficiency (recall) and purity (precision) analysis of ML Model PID
+///
+/// \author Michał Olędzki <mioledzk@cern.ch>
```
Comment (WildJellybear) [inline Tools/PIDML/pidMLBatchEffAndPurProducer.cxx]: my email is m.oledzki XD

**Review by WildJellybear — APPROVED** [review]
:D

File: Tools/PIDML/pidOnnxModel.h [hunk 5]
```diff
@@ -179,51 +169,63 @@ struct PidONNXModel {
...
+    float scaledY = (track.y() - mScalingParams.at("fY").first) / mScalingParams.at("fY").second;
+    float scaledZ = (track.z() - mScalingParams.at("fZ").first) / mScalingParams.at("fZ").second;
+    float scaledAlpha = (track.alpha() - mScalingParams.at("fAlpha").first) / mScalingParams.at("fAlpha").second;
+    float scaledTPCNClsShared = (static_cast<float>(track.tpcNClsShared()) - mScalingParams.at("fTPCNClsShared").first) / mScalingParams.at("fTPCNClsShared").second;
+    float scaledDcaXY = (track.dcaXY() - mScalingParams.at("fDcaXY").first) / mScalingParams.at("fDcaXY").second;
+    float scaledDcaZ = (track.dcaZ() - mScalingParams.at("fDcaZ").first) / mScalingParams.at("fDcaZ").second;
 
-    return inputValues;
-  }
+    inputValues.insert(inputValues.end(), {track.p(), track.pt(), track.px(), track.py(), track.pz(), static_cast<float>(track.sign()), scaledX, scaledY, scaledZ, scaledAlpha, static_cast<float>(track.trackType()), scaledTPCNClsShared, scaledDcaXY, scaledDcaZ});
```
Comment (saganatt) [inline Tools/PIDML/pidOnnxModel.h:214]: Czy w takiej samej kolejności model miał dane przy trenowaniu?

File: Tools/PIDML/pidOnnxModel.h [hunk 6]
```diff
@@ -179,51 +169,63 @@ struct PidONNXModel {
...
 
-  // FIXME: Temporary solution, new networks will have sigmoid layer added
-  float sigmoid(float x)
```
Comment (saganatt) [inline Tools/PIDML/pidOnnxModel.h:217]: Może sigmoida na razie nie będziemy usuwać

**Review by saganatt — CHANGES_REQUESTED** [review]

**Comment by saganatt** [comment]
Poza tymi 2 uwagami dla mnie spoko.
I tak bym poczekała z mergem, aż na pewno model podziała na Run 3.

File: Tools/PIDML/pidOnnxModel.h (hunk 5 above)
Comment (mytkom) [reply Tools/PIDML/pidOnnxModel.h:214]: Tak, generujemy columns_for_training.json podczas trenowania:
```json
{
    "columns_for_training": [
        "fTPCSignal",
        "fTRDSignal",
        "fTRDPattern",
        "fTOFSignal",
        "fBeta",
        "fP",
        "fPt",
        "fPx",
        "fPy",
        "fPz",
        "fSign",
        "fX",
        "fY",
        "fZ",
        "fAlpha",
        "fTrackType",
        "fTPCNClsShared",
        "fDcaXY",
        "fDcaZ"
    ]
}
```
Kolejność jest na jego podstawie.

**Review by saganatt — APPROVED** [review]

**Comment by saganatt** [comment]
For some reason clang format wants additional indentation. Please check you have all brackets etc. set right.

To be merged after successful tests

**Comment by mytkom** [comment]
I will close this PR and make 3 separate just so I skip conflicts resolving and everything would be cleaner (it means 3 PRs to original O2Physics repo from this fork, but each one would be logically separated)

**Comment by mytkom** [comment]
That's it.

**Pull request closed without merging**